Atom Matrix用のインストールガイド

ksbmyk opened the pull request:
Win（WSL）とMacでインストールガイドを作りました。
この通りで動くか複数人でためしたいです 🙏 
説明を足した方がいいなどあればコメントいただきたいです。

http://localhost:4000/guides/esp32/ から確認できます（まだ http://localhost:4000/guides/ からはリンクを貼っていません）


2025/10/19 追記
インストールガイドといいつつ、ワークショップのカリキュラム全て入れています
（時間短縮のために、今回のワークショップの内容をやる専用のガイドという形になっています）

ksbmyk committed:
add setup guid for esp32

emorima commented on guides/esp32/wsl_setup.markdown:
```diff
@@ -0,0 +1,285 @@
+
+## 1. WSL環境の準備
+### 1.1 WSLの起動
+Windowsターミナルを開きます（スタートメニューで「Terminal」と検索）
```
:memo: 

Terminalと入力途中で「ターミナル」(Windows Power Shell) と「コマンド プロンプト」が切り替わるのですが、結局 wsl にログインするので、どちらでもOKでした。

emorima replied:
Rails Girls のインストールガイドで、 PowerShell を使っているのと、
usbipd-winのインストールでも PowerShell を使うので、「Terminal」と検索させずに、
ガイド全体で「Powershell」で統一させてしまう方がよさそうに思うのですが、いかがでしょうか？

ksbmyk replied:
なるほどそういう違いが。Windowsでのターミナルは「Powershell」を使ってもらうようにしましょう！

emorima commented on guides/esp32/wsl_setup.markdown:
````diff
@@ -0,0 +1,285 @@
+git clone --recursive https://github.com/picoruby/R2P2-ESP32.git
+cd R2P2-ESP32
+```
+* :memo: なぜ --recursive が必要？
````
`:memo:` 使うために、Gemfileに `jemoji` の追加が必要そうです！

ksbmyk replied:
うっかりしてました。一旦ユニコードで

emorima commented on guides/esp32/wsl_setup.markdown:
````diff
@@ -0,0 +1,285 @@
+sudo apt-get install git wget flex bison gperf python3 python3-pip python3-venv cmake ninja-build ccache libffi-dev libssl-dev dfu-util libusb-1.0-0
+```
+
+1.3 ESP-IDFのインストール
````
```suggestion
### 1.3 ESP-IDFのインストール
```

asya81 commented on guides/esp32/irb.markdown:
````diff
@@ -0,0 +1,108 @@
+irbを起動してください
+
+```
+b = GPIO.new(39, GPIO::IN, GPIO::PULL_UP)
````
@ksbmyk 
こちらを実行したところ、以下のエラーになりました。
どこに問題がありそうでしょうか？
```
E (69902) gpio: gpio_pullup_en(78): GPIO number error (input-only pad has no internal PU)
```

ksbmyk replied:
@asya81 
ありがとうございます！ GPIO 39は入力専用ピンなので、 `GPIO::PULL_UP` 指定することができないのが正常だと思います（なぜか私の手元ではエラーになってくれなかったのですが…）
そもそも指定しなくても大丈夫なので以下でためしてもらえますか？ 🙏 

`b = GPIO.new(39, GPIO::IN)`

ksbmyk committed:
GPIO 39の初期化を修正（一部環境でのプルアップエラーを回避）

ESP32のGPIO 39は入力専用ピンで内部プルアップ非対応のため、
GPIO::PULL_UPを指定すると環境によってはエラーが発生する。
ATOM Matrixは外部プルアップ抵抗を実装済みのため、内部プルアップは不要。

ksbmyk committed:
add button doc

ksbmyk committed:
add gem before build

ksbmyk committed:
remove button

ksbmyk committed:
add led doc

ksbmyk committed:
add accel doc

ksbmyk committed:
add advanced doc

ksbmyk committed:
add curriculum for index

ksbmyk committed:
use PowerShell

ksbmyk committed:
add Heading

ksbmyk committed:
use unicode

ksbmyk committed:
gemの記述分離

ksbmyk committed:
ガイドの構成を変更

ksbmyk committed:
番号の振り直し、説明の修正

ksbmyk commented:
一旦マージして触ってもらうことにします〜
